tests/e2e/auth_test.go

```diff
@@ -45,6 +49,64 @@ func (t *apiKeyTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	return http.DefaultTransport.RoundTrip(req)
 }
 
+func TestKeyManagerKeys(t *testing.T) {
```

Comment:
I was expecting 3 different tests scenarios to validate TLS auth works as expected:
1. Using not certificate should not be rejected (Done)
2. Using valid certificate should be accepted (Done)
3. Using invalid certificate should be rejected (Missing)

Reply:
Missing bit when I picked from the old branch